theano/configdefaults.py

```diff
@@ -1238,21 +1252,20 @@ def default_blas_ldflags():
                                           "mk2_rt"]])
 
         # Anaconda
-        if "Anaconda" in sys.version or "Continuum" in sys.version:
-            # If the "mkl-service" conda package (available
-            # through Python package "mkl") is installed and
+        if "Continuum Analytics" in sys.version:
+            # If the "mkl-service" conda package is installed and
             # importable, then the libraries (installed by conda
-            # package "mkl-rt") are actually available.  Using
-            # "conda install mkl" will install both, as well as
-            # optimized versions of numpy and scipy.
+            # package "mkl") are actually available.  Using
+            # "conda install mkl-service" will install both,
+            # as well as optimized versions of numpy and scipy.
             try:
                 import mkl  # noqa
             except ImportError as e:
-                _logger.info('Conda mkl is not available: %s', e)
+                _logger.warning('Conda mkl is not available, use `conda install mkl-service`: %s', e)
             else:
                 # This branch is executed if no exception was raised
                 if sys.platform == "win32":
-                    lib_path = os.path.join(sys.prefix, 'DLLs')
+                    lib_path = os.path.join(sys.prefix, 'Library', 'bin')
                     flags = ['-L"%s"' % lib_path]
```

Comment:
Hello !

Not sure if the problem comes exactly from here, but I have tried to install this pull request on Windows 8 an then execute theano/misc/check_blas.py to test it, but I got some errors related to os.listdir().

First, the theano version:
```
(python2) C:\Users\HPPC\mila\test>python -c "import theano; print theano.__version__"
0.9.0dev4.dev-0fc36d4db9a73e921a6b1afe15528c952de65093
```

Then, the test:

```
(python2) C:\Users\HPPC\mila\test>python -m theano.misc.check_blas > conda-pull-request-5207-theano-
python2.log 2> conda-pull-request-5207-theano-python2-stderr.log
```

And the error printed in conda-pull-request-5207-theano-python2-stderr.log:

```
ERROR (theano.gof.cmodule): [Error 123] La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte: '"C:\\donnees\\programmes\\Anaconda3\\envs\\python2\\Library\\bin"/*.*'
Traceback (most recent call last):
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\misc\check_blas.py", line 300, in <module>
    order=options.order)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\misc\check_blas.py", line 67, in execute
    f = theano.function([], updates=[(c, 0.4 * c + .8 * T.dot(a, b))])
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\compile\function.py", line 326, in function
    output_keys=output_keys)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\compile\pfunc.py", line 486, in pfunc
    output_keys=output_keys)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\compile\function_module.py", line 1784, in orig_function
    defaults)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\compile\function_module.py", line 1651, in create
    input_storage=input_storage_lists, storage_map=storage_map)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\link.py", line 699, in make_thunk
    storage_map=storage_map)[:3]
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\vm.py", line 1057, in make_all
    impl=impl))
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\op.py", line 924, in make_thunk
    no_recycling)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\op.py", line 828, in make_c_thunk
    output_storage=node_output_storage)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\cc.py", line 1190, in make_thunk
    keep_lock=keep_lock)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\cc.py", line 1131, in __compile__
    keep_lock=keep_lock)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\cc.py", line 1589, in cthunk_factory
    key=key, lnk=self, keep_lock=keep_lock)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\cmodule.py", line 1155, in module_from_key
    module = lnk.compile_cmodule(location)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\cc.py", line 1490, in compile_cmodule
    lib_dirs=self.lib_dirs(),
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\cc.py", line 1085, in lib_dirs
    ret += x.c_lib_dirs()
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\tensor\blas.py", line 504, in c_lib_dirs
    return ldflags(libs=False, libs_dir=True)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\tensor\blas.py", line 386, in ldflags
    include_dir=include_dir)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\gof\utils.py", line 258, in rval
    val = f(*args, **kwargs)
  File "C:\donnees\programmes\Anaconda3\envs\python2\lib\site-packages\theano\tensor\blas.py", line 423, in _ldflags
    for f in os.listdir(d):
WindowsError: [Error 123] La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte: '"C:\\donnees\\programmes\\Anaconda3\\envs\\python2\\Library\\bin"/*.*'
```

So, it seems os.listdir() called in theano/tensor/blas.py is angry, and I suspects it is because of the quotes around the path. So I run the following script:

```
from __future__ import print_function
import sys, os, platform

print('Platform:', platform.platform())
print('Python version:', sys.version_info[0:])
print('Python sys.prefix:', sys.prefix)
print()
try:
	path_content = os.listdir(sys.prefix)
	print('os.listdir() succeeds with sys.prefix.')
except Exception as e:
	print('os.listdir() fails with sys.prefix.')
	print(e)
print()
try:
	path_content = os.listdir('"%s"' % sys.prefix)
	print('os.listdir() succeeds with sys.prefix surrounded with quotes.')
except Exception as e:
	print('os.listdir() fails with sys.prefix surrounded with quotes.')
	print(e)
```

And I got this output:
```
Platform: Windows-8.1-6.3.9600
Python version: (2, 7, 12, 'final', 0)
Python sys.prefix: C:\donnees\programmes\Anaconda3\envs\python2

os.listdir() succeeds with sys.prefix.

os.listdir() fails with sys.prefix surrounded with quotes.
[Error 123] La syntaxe du nom de fichier, de rÚpertoire ou de volume est incorrecte: '"C:\\donnees\\
programmes\\Anaconda3\\envs\\python2"/*.*'
```

So it seems it is indeed because of the quotes, and I think that the path passed to os.listdir() is generated at this line of theano/configdefaults.py (```flags = ['-L"%s"' % lib_path]```).

Maybe the call to os.listdir() should be modified in theano/tensor/blas.py by replacing
```for f in os.listdir(d):```
with:
```for f in os.listdir(d.strip('"')):```

?